src/common/hist_util.h

```diff
@@ -115,10 +117,24 @@ struct HistCutMatrix {
   Monitor monitor_;
 };
 
+/*!
+ * \brief A container that holds the device sketches across all
+ *  sparse page batches which are distributed to different devices.
+ *  As sketches are aggregated by column, the atomic flag simulates
+ *  a lock in user land when multiple devices could be pushing
+ *  sketch summary for the same column across distinct rows.
+ */
+struct SketchContainer {
```

Comment:
I wonder how this works in a distributed/multi-node setup. Have you tried to run this with multiple workers?

Reply:
is this specific to this code fragment or a question in general?

typically, in a distributed setup, wouldn't the dmatrix be appropriately created by splitting the rows across multiple workers, with different parts of the file going to different workers. at this phase, this only deals with the set of rows it has been handed over. 

i haven't tried this in a multi-node/distributed training environment.

Reply:
Yeah I guess the question is not specific to this code fragment. But if in a distributed setup each worker only works with its local data shard, perhaps in external memory mode we also don't need to  accumulate quantiles across all the batches? If a single batch is representative enough, maybe that's all we need? Just a thought, not sure if it'll work out. :)

Reply:
This structure definitely belongs to the internals of GPUSketcher, not to its public interface. Could you move it there?